### PR TITLE
Fixed notebook, dataset, docker config, removed yacs

### PR DESCRIPTION
**Pull request by 86sanj**
# Peer Review Information

This commit includes following changes:
1. Refract SynDetection2D to fix the notebook.
2. Removed yacs lib
3. Included docker build fixes (https://github.com/Unity-Technologies/datasetinsights/pull/150)
4. Update README to reflect the current high level API with instructions to show how to use this package

Requirement: https://github.com/Unity-Technologies/datasetinsights/issues/143

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: notebooks/Perception_Statistics.ipynb [hunk 1]
```diff
@@ -13,11 +13,11 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 8,
+   "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
-    "data_root = \"/data/<GUID>\""
+    "data_root = \"/Users/sanjay.vishwakarma/data_test\""
```
Comment (BlairLee) [inline notebooks/Perception_Statistics.ipynb]: Please clean the notebook (remove the personal path and all outputs) in this PR.

File: README.md [hunk 2]
```diff
@@ -18,13 +18,28 @@ We provide a sample [notebook](notebooks/SynthDet_Statistics.ipynb) to help you
...
 
-[Download Dataset](https://datasetinsights.readthedocs.io/en/latest/datasetinsights.commands.html#datasetinsights-commands-download)
+[CLI](https://datasetinsights.readthedocs.io/en/latest/datasetinsights.commands.html#datasetinsights-commands-download)
 
 ```bash
 datasetinsights download \
   --source-uri=<xxx> \
   --output=$HOME/data
 ```
+[Programmatically](https://datasetinsights.readthedocs.io/en/latest/datasetinsights.io.downloader.html#module-datasetinsights.io.downloader.gcs_downloader)
```
Comment (BlairLee) [inline README.md:26]: Do we need to add this instruction here?

Comment (86sanj) [reply README.md:26]: Yes, Please refer release checklist: https://github.com/Unity-Technologies/datasetinsights/issues/143
"Update README to reflect the current high level API."

File: README.md [hunk 3]
```diff
@@ -12,19 +12,34 @@ Dataset Insights maintains a pip package for easy installation. It can work in a
 
 ### Dataset Statistics
 
-We provide a sample [notebook](notebooks/SynthDet_Statistics.ipynb) to help you get started with dataset statistics for the [SynthDet](https://github.com/Unity-Technologies/SynthDet) project. We plan to support other sample Unity projects in the future.
+We provide a sample [notebook](notebooks/Perception_Statistics.ipynb) to help you load synthetic datasets generated from the [Perception package](https://github.com/Unity-Technologies/com.unity.perception) and visualize dataset statistics. We plan to support other sample Unity projects in the future.
```
Comment (romerocesar) [inline README.md]: "generated by" or "generated using"

File: README.md [hunk 4]
```diff
@@ -12,19 +12,34 @@ Dataset Insights maintains a pip package for easy installation. It can work in a
...
 
 ### Dataset Download
 
-Dataset download provides tools to download datasets from HTTP(s), GCS and Unity simulation project . You can run `download` command:
+Dataset download provides tools to download datasets from HTTP(s), GCS and Unity simulation project . You can run `download` command from cli or use api to download the dataset.
```
Comment (romerocesar) [inline README.md]: This makes it sond like "dataset download is a separate pkg which is confusing". Also missing articles "the command" and "the API"

File: README.md [hunk 5]
```diff
@@ -12,19 +12,34 @@ Dataset Insights maintains a pip package for easy installation. It can work in a
...
+>>> downloader = UnitySimulationDownloader(access_token=access_token)
+>>> downloader.download(source_uri=source_uri, output=data_root)
+
+>>> downloader = GCSDatasetDownloader()
+>>> downloader.download(source_uri=source_uri, output=data_root)
+
+>>> downloader = HTTPDatasetDownloader()
+>>> downloader.download(source_uri=source_uri, output=data_root)
+
+```
```
Comment (romerocesar) [inline README.md:42]: the first line should be \```python3 to get proper syntax highlighting, and the following lines should not include `>>>`

File: datasetinsights/datasets/synthetic.py [hunk 6]
```diff
@@ -72,3 +85,182 @@ def read_bounding_box_2d(annotation, label_mappings=None):
         bboxes.append(box)
 
     return bboxes
+
+
+class SynDetection2D:
+    """Synthetic dataset for 2D object detection.
+
+    During the class instantiation, it would check whether the data files
```
Comment (romerocesar) [inline datasetinsights/datasets/synthetic.py:93]: what would check?

**Review by romerocesar — DISMISSED** [review]

File: datasetinsights/datasets/synthetic.py (hunk 6 above)
Comment (86sanj) [reply datasetinsights/datasets/synthetic.py:93]: Its previous class without referring dataset base class.

**Review by BlairLee — DISMISSED** [review]

**Review by BlairLee — DISMISSED** [review]

File: README.md [hunk 7]
```diff
@@ -12,19 +10,36 @@ Dataset Insights maintains a pip package for easy installation. It can work in a
```
Comment (BlairLee) [inline README.md:10]: I think >= 3.7 is too vague. We only test it supports 3.7 and 3.8 but not 3.9. Can we specify in this way: Python 3 (3.7 and 3.8)

Comment (86sanj) [reply README.md:10]: I am not sure whether pining to a particular version (3.7 and 3.8) would be good approach.

**Review by BlairLee — APPROVED** [review]

**Review by Saurav-D — APPROVED** [review]

File: datasetinsights/datasets/synthetic.py [hunk 8]
```diff
@@ -72,3 +85,182 @@ def read_bounding_box_2d(annotation, label_mappings=None):
...
+        """ Preprocess dataset inside data_path and un-archive if necessary.
+
+        Args:
+            data_path (str): Path where dataset is stored.
+
+        Return:
+            Path of the dataset files.
+        """
+        archive_file = Path(data_path) / SynDetection2D.ARCHIVE_FILE
+        if archive_file.exists():
```
Comment (adason) [inline datasetinsights/datasets/synthetic.py:238]: @86sanj Why do we need to unarchive files for the dataset generated by perception package?

File: datasetinsights/datasets/synthetic.py [hunk 9]
```diff
@@ -72,3 +85,182 @@ def read_bounding_box_2d(annotation, label_mappings=None):
         bboxes.append(box)
 
     return bboxes
+
+
+class SynDetection2D:
```
Comment (adason) [inline datasetinsights/datasets/synthetic.py:90]: @86sanj This looks like an object that will confuse users with the pytorch dataset class. Can we support visual inspection in the `preception_statistics` notebook without this class?

File: Dockerfile [hunk 10]
```diff
@@ -13,7 +13,8 @@ RUN apt-get update \
     && ln -s /usr/bin/python3.7 /usr/local/bin/python
 
 # Pin setuptools to 49.x.x until this [issue](https://github.com/pypa/setuptools/issues/2350) is fixed.
-RUN python -m pip install --upgrade pip poetry==1.0.10 setuptools==49.6.0
+RUN python -m pip install --upgrade pip poetry==1.0.10 setuptools==49.6.0 -U pip cryptography==3.3.2
+# pin cryptography to 3.3.2 until this (https://github.com/Azure/azure-cli/issues/16858) is fixed.
```
Comment (adason) [inline Dockerfile:17]: @86sanj How is datasetinsights docker build related to azure-cli issues? What's the root cause of pinning `cryptography==3.3.2`

File: datasetinsights/constants.py [hunk 11]
```diff
@@ -4,26 +4,13 @@
...
-GCS_BUCKET = "thea-dev"
 GCS_BASE_STR = "gs://"
 HTTP_URL_BASE_STR = "http://"
 HTTPS_URL_BASE_STR = "https://"
 LOCAL_FILE_BASE_STR = "file://"
 
-# This is a hack on yacs config system, as it does not allow null values
-# in configs. They are working on supporting null values in config
-# https://github.com/rbgirshick/yacs/pull/18.
 NULL_STRING = "None"
```
Comment (adason) [inline datasetinsights/constants.py:12]: This is only related to yacs and should be removed.

File: Dockerfile (hunk 10 above)
Comment (86sanj) [reply Dockerfile:17]: Issue: Latest version(3.4.6) cryptography released with adds rust as a dependency for building it. This breaks compatibility with the latest pip version. So suggestion was to pin at cryptography<3.4. 

Reference I should change to official site: https://github.com/pyca/cryptography/issues/5753.

**Pull request merged**